.github/workflows/compatibility-report.yml

```diff
@@ -0,0 +1,48 @@
+name: Generate Compatibility Report
+
+on:
+  workflow_dispatch:
+    inputs:
+      repo_target:
+        description: Generate the report for the local repo, the docs repo, or both.
+        required: true
+        type: choice
+        default: both
+        options:
+          - local
+          - docs
+          - both
+
+  pull_request:
+    paths:
+      - 'test/versioned/**/package.json'
+
+jobs:
+  local:
+    runs-on: ubuntu-latest
+    if:
+      github.event_name == 'pull_request' ||
+      (github.event_name == 'workflow_dispatch' &&
+      (input.repo_target == 'local' || input.repo_target == 'both'))
+    steps:
+      - uses: actions/checkout@v4
+      - uses: jaxxstorm/action-install-gh-release@71d17cb091aa850acb2a1a4cf87258d183eb941b
+        with:
+          repo: newrelic/newrelic-node-versions
+          platform: linux
+          arch: amd64
+          cache: enable
+      - run: |
+          nrversions -v -r . 2>status.log >./compatibility.md
```

Comment:
this is just going to get the versioned tests in this repo. we'd be omitting next.js and apollo plugin

Reply:
It should read the external repos configuration and pull those as well.

Reply:
when i wired that up `-r` was mutually exclusive from cloning repos. I just tested with `-r` and it just hangs now so something has changed

Reply:
☹️ we have some wonkiness around specifying the repo.

Reply:
In my view, `-r` should skip cloning of the main repo but still clone any repos defined in that repo under `/test/versioned-external/external-repos.js`.

Reply:
ok so yea `-r` is def not working. I'll PR with this fix

```diff --git a/main.go b/main.go
index 1ecaeff..ecf729d 100644
--- a/main.go
+++ b/main.go
@@ -271,6 +271,7 @@ func cloneRepos(repos []nrRepo, repoChan chan repoIterChan, wg *sync.WaitGroup)
 				repoDir:  repo.repoDir,
 				testPath: repo.testPath,
 			}
+			defer wg.Done()
```

However, it won't clone repos it doesn't have overrides for, that was not the intent

Reply:
https://github.com/newrelic/newrelic-node-versions/pull/17